doc/conf_ref.rst

```diff
@@ -525,7 +525,21 @@ configuration file by your distribution to override the DNF defaults.
     ``"/etc/dnf/vars", "/etc/yum/vars"``. See :ref:`variable files <varfiles-label>`
     in Configuration reference.
 
-.. _conf_repo_options-label:
+.. _exclude_from_weak-label:
+
+``exclude_from_weak``
+    :ref:`list <list-label>`
+
+    Prevent installing packages specified by a name or a glob and separated by a comma to satisfy recommends
+    or supplements dependency. Defaults to ``[]``.
+
+.. _exclude_from_weak_autodetect-label:
+
+``exclude_from_weak_autodetect``
+    :ref:`boolean <boolean-label>`
+
+    If enabled it will autodetect unmet weak dependencies (recommends or supplements) of installed packages and block
+    installation of packages satisfying unmet dependencies. Defaults to ``true``.
```

Comment:
I have couple of notes to this one, since it's very hard to find a description of this option that would be accurate yet easily understandable.

Firstly, my suggestion: If enabled, dnf will autodetect unmet weak dependencies (recommends or supplements) of already installed packages and won't install providers of these dependencies as weak dependencies (but they can still be installed to satisfy regular dependencies). Defaults to ``true``.

Explanation of the changes I suggested:
1. Minor improvements: add comma after "if enabled" and replace "it" for "dnf".
2. Add "already" before "installed packages" to make it immediately clear that we are talking about packages that are already on the system and not packages that are currently being installed.
3. Replace "block installation of" with "won't install", because "block" might seem unnecessarily radical (like the whole transaction will be blocked or something).
4. The "packages satisfying unmet dependencies" seems very confusing to me. I'm not sure if my suggestion is good enough. Other alternative might be to replace the ending with: "won't install packages that satisfy these autodetected weak dependencies as weak dependencies"
5. Additional explanatory sentence into parentheses afterwards just to make it really clear.

Let me know if this seems ok to you. Ideally, I'd like to then ask someone else to read the description, because I'm biased at this point (and my English is not that good).

Reply:
Thanks, I used your suggested version with a minor change. I skipped "already" from point 2. Somehow I feel that it is redundant.